In [1]:
import sys
sys.path.append(r'C:\Users\Acer\Documents\IC')
import numpy as np

In [31]:
filename = 'all_solutions.npz'
data = np.load(filename)
gurobi_values = data["all_solutions"]
print(gurobi_values)

gurobi_10 = gurobi_values[0]
gurobi_30 = gurobi_values[1]
gurobi_100 = gurobi_values[2]

[ 7316203.9254538  19978014.71339107 70598781.54841062]


In [14]:
def percentage_difference(value1, value2):
    if value1 == value2:
        print("Both values are equal.")
        return
        
    difference = abs(value1 - value2)
    average = (value1 + value2) / 2
    percentage_diff = (difference / average) * 100
        
    if value1 > value2:
        print(f"{value1} is {percentage_diff:.2f}% greater than {value2}.")
    else:
        print(f"{value1} is {percentage_diff:.2f}% less than {value2}.")

# Problem size 10

## set 1

In [36]:
filename = '../sets1/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

min_a = np.min(a)
min_b = np.min(b)

print(min_a)
print(min_b)

7360628.890699309
7370941.559357028


In [37]:
percentage_difference(gurobi_10, min_a)
percentage_difference(gurobi_10, min_b)

7316203.925453799 is 0.61% less than 7360628.890699309.
7316203.925453799 is 0.75% less than 7370941.559357028.


## set 2

In [34]:
filename = '../sets2/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

d = data[3, :]

min_d = np.min(d)

print(min_d)

7375855.950176503


In [35]:
percentage_difference(gurobi_10, min_d)

7316203.925453799 is 0.81% less than 7375855.950176503.


## set 3

In [38]:
filename = '../sets3/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

min_a = np.min(a)
min_b = np.min(b)

print(min_a)
print(min_b)

7377593.709733897
7358891.131141915


In [39]:
percentage_difference(gurobi_10, min_a)
percentage_difference(gurobi_10, min_b)

7316203.925453799 is 0.84% less than 7377593.709733897.
7316203.925453799 is 0.58% less than 7358891.131141915.


# Problem size 30

## set 1

In [40]:
filename = '../problem_30/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

min_a = np.min(a)
min_b = np.min(b)

print(min_a)
print(min_b)

19959248.326481618
19959248.326481618


In [41]:
percentage_difference(gurobi_30, min_a)
percentage_difference(gurobi_30, min_b)

19978014.71339107 is 0.09% greater than 19959248.326481618.
19978014.71339107 is 0.09% greater than 19959248.326481618.


## set 2

In [42]:
filename = '../problem_30/FX_values_2_0.npz'
data = np.load(filename)
data = data["FX_values_"]

d = data[3, :]

min_d = np.min(d)

print(min_d)

19959248.326481618


In [43]:
percentage_difference(gurobi_30, min_d)

19978014.71339107 is 0.09% greater than 19959248.326481618.


## set 3

In [44]:
filename = '../problem_30/FX_values_3_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

min_a = np.min(a)
min_b = np.min(b)

print(min_a)
print(min_b)

19980955.249337316
20032917.97972587


In [45]:
percentage_difference(gurobi_30, min_a)
percentage_difference(gurobi_30, min_b)

19978014.71339107 is 0.01% less than 19980955.249337316.
19978014.71339107 is 0.27% less than 20032917.97972587.


# Problem size 100

## set 1

In [50]:
filename = '../problem_100/FX_values_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

min_a = np.min(a)
min_b = np.min(b)

print(min_a)
print(min_b)

71523794.09009075
71476182.51642042


In [51]:
percentage_difference(gurobi_100, min_a)
percentage_difference(gurobi_100, min_b)

70598781.54841062 is 1.30% less than 71523794.09009075.
70598781.54841062 is 1.24% less than 71476182.51642042.


## set 2

In [48]:
filename = '../problem_100/FX_values_2_0.npz'
data = np.load(filename)
data = data["FX_values_"]

d = data[3, :]

min_d = np.min(d)

print(min_d)

71524504.06173822


In [49]:
percentage_difference(gurobi_100, min_d)

70598781.54841062 is 1.30% less than 71524504.06173822.


## set 3

In [46]:
filename = '../problem_100/FX_values_3_0.npz'
data = np.load(filename)
data = data["FX_values_"]

a = data[0, :]
b = data[1, :]

min_a = np.min(a)
min_b = np.min(b)

print(min_a)
print(min_b)

71904221.77132784
72510693.51183414


In [47]:
percentage_difference(gurobi_100, min_a)
percentage_difference(gurobi_100, min_b)

70598781.54841062 is 1.83% less than 71904221.77132784.
70598781.54841062 is 2.67% less than 72510693.51183414.


## Problem Size 10

Set 1:
- min_a: 0.61% maior que Gurobi
- min_b: 0.75% maior que Gurobi

Set 2:
- min_d: 0.81% maior que Gurobi

Set 3:
- min_a: 0.84% maior que Gurobi
- min_b: 0.58% maior que Gurobi

## Problem Size 30

Set 1:
- min_a: 0.09% menor que Gurobi
- min_b: 0.09% menor que Gurobi

Set 2:
- min_d: 0.09% maior que Gurobi

Set 3:
- min_a: 0.01% menor que Gurobi
- min_b: 0.27% menor que Gurobi

## Problem Size 100

Set 1:
- min_a: 1.30% maior que Gurobi
- min_b: 1.24% maior que Gurobi

Set 2:
- min_d: 1.30% maior que Gurobi

Set 3:
- min_a: 1.83% maior que Gurobi
- min_b: 2.67% maior que Gurobi